clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1020,6 +1051,23 @@ private void killConsumptionTaskForNonCurrentVersions() {
     LOGGER.info("Finished killing the following ingestion tasks: {}", nonCurrentVersions);
   }
 
+  private void scanAndCloseIdleConsumptionTasks() {
+    try {
+      LOGGER.info("Number of ingestion tasks before cleaning: {}", topicNameToIngestionTaskMap.size());
+      for (Map.Entry<String, StoreIngestionTask> entry: topicNameToIngestionTaskMap.entrySet()) {
+        String topicName = entry.getKey();
+        StoreIngestionTask task = entry.getValue();
+        if (task.isIdleOverThreshold()) {
+          LOGGER.info("Found idle task for topic {}, shutting it down.", topicName);
+          shutdownIdleIngestionTask(topicName);
+        }
+      }
+      LOGGER.info("Number of active ingestion tasks after cleaning: {}", topicNameToIngestionTaskMap.size());
```

Comment:
nit: can we print only if we clean any SIT, depending on how frequent you plan to run it can generate unnecessary excessive logging.

Reply:
Going to run it at most once every 10 minutes.